invenio_rdm_records/links/models.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Secret links for sharing access to records."""
+
+import uuid
+from copy import deepcopy
+from datetime import date, datetime
+
+from flask import current_app, url_for
+from invenio_access.models import User
+from invenio_db import db
+from itsdangerous import BadData
+from sqlalchemy_utils import EncryptedType, UUIDType
+
+from .permissions import LinkNeed
+from .serializers import SecretLinkSerializer, TimedSecretLinkSerializer
+from .signals import link_created, link_revoked
+
+SUPPORTED_DIGEST_ALGORITHMS = ("HS256", "HS512")
+
+
+def secret_key():
+    """Return secret key as bytearray."""
+    return current_app.config["SECRET_KEY"].encode("utf-8")
+
+
+class SecretLink(db.Model):
+    """Secret links for sharing permissions on records."""
+
+    __tablename__ = "rdm_records_secret_links"
+
+    id = db.Column(UUIDType, primary_key=True, default=uuid.uuid4)
+    """Secret link ID."""
+
+    token = db.Column(
+        EncryptedType(type_in=db.Text, key=secret_key), nullable=False
+    )
+    """Secret token for link (should be stored encrypted)."""
+
+    owner_user_id = db.Column(
+        db.Integer, db.ForeignKey(User.id), nullable=False, default=None
+    )
+    """Owner's user ID."""
+
+    owner = db.relationship(User, foreign_keys=[owner_user_id])
+    """Relationship to user"""
+
+    created = db.Column(
+        db.DateTime, nullable=False, default=datetime.utcnow, index=True
+    )
+    """Creation timestamp."""
+
+    expires_at = db.Column(db.DateTime, nullable=True)
+    """Expiration date."""
+
+    revoked_at = db.Column(db.DateTime, nullable=True, index=True)
+    """Revocation timestamp."""
+
+    title = db.Column(db.String, nullable=False, default="")
```

Comment:
Comment: I would remove title and description - it made sense in the zenodo-accessrequest but not sure it makes sense here.